### PR TITLE
Preserve quotedness for Identifier when formatting Query

### PR DESCRIPTION
**Pull request by v-jizhang**
Cherry-pick of https://github.com/trinodb/trino/pull/80

Presto doesn't maintain the quotedness of an identifier when
using SqlQueryFormatter so it results in throwing parsing error
when we run prepare query of the syntax
[https://github.com/prestodb/presto/issues/10739]. This PR
solves that above issue.

Co-authored-by: praveenkrishna <praveenkrishna@tutanota.com>

Test plan - Added tests

```
== RELEASE NOTES ==

General Changes
* Preserve quotedness for Identifier when formatting Query

```

### CODE REVIEW

**Review by swapsmagic — CHANGES_REQUESTED** [review]

**Comment by v-jizhang** [comment]
The failed test is unrelated.

File: presto-main/src/test/java/com/facebook/presto/server/TestHttpRequestSessionContext.java [hunk 1]
```diff
@@ -161,7 +161,7 @@ public void testPreparedStatementsSpecialCharacters()
                         .put(PRESTO_LANGUAGE, "zh-TW")
                         .put(PRESTO_TIME_ZONE, "Asia/Taipei")
                         .put(PRESTO_CLIENT_INFO, "null")
-                        .put(PRESTO_PREPARED_STATEMENT, "query1=select * from tbl:ns")
+                        .put(PRESTO_PREPARED_STATEMENT, "query1=select * from \"tbl:ns\"")
```
Comment (swapsmagic) [inline presto-main/src/test/java/com/facebook/presto/server/TestHttpRequestSessionContext.java:164]: What is the reason for having this query with delimiter? If it's needed, we should have it generated using the parser so if in future we change the delimiter, this test won't break.

**Review by swapsmagic — APPROVED** [review]
one minor comment, rest looks good.

**Comment by swapsmagic** [comment]
Also please get the failed test pass by re-running the tests.

File: presto-main/src/test/java/com/facebook/presto/server/TestHttpRequestSessionContext.java (hunk 1 above)
Comment (v-jizhang) [reply presto-main/src/test/java/com/facebook/presto/server/TestHttpRequestSessionContext.java:164]: Please the commit message [here](https://github.com/prestodb/presto/pull/17186/commits/f80abb112d62cdf61a34c804cada6ecffa5c6885)

**Comment by rubenssoto** [comment]
Is there any chance of being merged in the next version? Very important fix for us.

Thank you!

**Comment by jbapple** [comment]
Looks like this is missing one followup from @v-jizhang

**Comment by v-jizhang** [comment]
@jbapple All tests passed.

**Review by zhenxiao — APPROVED** [review]
looks good

**Pull request merged**